pkg/lint/package.go

```diff
@@ -318,3 +186,96 @@ func (cl *ContextLoader) filterDuplicatePackages(pkgs []*packages.Package) []*pa
 
 	return retPkgs
 }
+
+func (l *PackageLoader) debugPrintLoadedPackages(pkgs []*packages.Package) {
+	l.debugf("loaded %d pkgs", len(pkgs))
+	for i, pkg := range pkgs {
+		var syntaxFiles []string
+		for _, sf := range pkg.Syntax {
+			syntaxFiles = append(syntaxFiles, pkg.Fset.Position(sf.Pos()).Filename)
+		}
+		l.debugf("Loaded pkg #%d: ID=%s GoFiles=%s CompiledGoFiles=%s Syntax=%s",
+			i, pkg.ID, pkg.GoFiles, pkg.CompiledGoFiles, syntaxFiles)
+	}
+}
+
+func (l *PackageLoader) prepareBuildContext() {
+	// Set GOROOT to have working cross-compilation: cross-compiled binaries
+	// have invalid GOROOT. XXX: can't use runtime.GOROOT().
+	goroot := l.goenv.Get(goutil.EnvGoRoot)
+	if goroot == "" {
+		return
+	}
+
+	os.Setenv(string(goutil.EnvGoRoot), goroot)
+	build.Default.GOROOT = goroot
+	build.Default.BuildTags = l.cfg.Run.BuildTags
+}
+
+func (l *PackageLoader) buildArgs() []string {
+	if len(l.args) == 0 {
+		return []string{"./..."}
+	}
+
+	var retArgs []string
+	for _, arg := range l.args {
+		if strings.HasPrefix(arg, ".") || filepath.IsAbs(arg) {
+			retArgs = append(retArgs, arg)
+		} else {
+			// go/packages doesn't work well if we don't have the prefix ./ for local packages
+			retArgs = append(retArgs, fmt.Sprintf(".%c%s", filepath.Separator, arg))
+		}
+	}
+
+	return retArgs
+}
+
+func (l *PackageLoader) makeBuildFlags() []string {
+	var buildFlags []string
+
+	if len(l.cfg.Run.BuildTags) != 0 {
+		// go help build
+		buildFlags = append(buildFlags, "-tags", strings.Join(l.cfg.Run.BuildTags, " "))
+		l.log.Infof("Using build tags: %v", l.cfg.Run.BuildTags)
+	}
+
+	if l.cfg.Run.ModulesDownloadMode != "" {
+		// go help modules
```

Comment:
Looks like we should move this comment right above the line in `Run.Validate`:
```
allowedMods := []string{"mod", "readonly", "vendor"}
```

Because it previously was there https://github.com/golangci/golangci-lint/blob/8f3027c631ffdf590af5d930af2c89726d1f16f3/pkg/lint/load.go#L137-L138.

Reply:
`go help modules` is related to both: the values and the flag `-mod=`.